smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/OutgoingKafkaRecord.java

```diff
@@ -198,12 +198,14 @@ public String key() {
 
     @Override
     public OutgoingKafkaRecord<K, T> withMetadata(Iterable<Object> metadata) {
+        // TODO this adds the entire provided Iterable<Object> as a single datum in the existing Metadata
```

Comment:
is this something to handle in this PR?

Reply:
I noticed this while writing the test, but couldn't quite figure out a proper solution, so I decided to at least leave a TODO. I still need to wrap my head around the entire `Metadata` thing.